tools/Explicit.NuGet.Versions/Program.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using Ionic.Zip;
+
+namespace Explicit.NuGet.Versions
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+	        var packageDiscoveryDirectory = Path.Combine(Environment.CurrentDirectory, args[0]);
+	        var packageDiscoverDirectoryInfo = new DirectoryInfo(packageDiscoveryDirectory);
+	        var packageMetaData = ReadNuspecFromPackages(packageDiscoverDirectoryInfo);
+	        UpdateNuspecManifestContent(packageMetaData, args[1]);
+	        WriteNuspecToPackages(packageMetaData);
+		}
+
+	    private static void WriteNuspecToPackages(Dictionary<string, NuspecContentEntry> packageMetaData)
+	    {
+		    foreach (var packageFile in packageMetaData.ToList())
+		    {
+			    using (var zipFile = ZipFile.Read(packageFile.Key))
+			    {
+				    zipFile.UpdateEntry(packageFile.Value.EntryName, packageFile.Value.Contents);
+				    zipFile.Save();
+			    }
+		    }
+	    }
+
+	    private static void UpdateNuspecManifestContent(Dictionary<string, NuspecContentEntry> packageMetaData, string dependencyNugetId)
+	    {
+		    foreach (var packageFile in packageMetaData.ToList())
+		    {
+			    var nuspecXmlDocument = new XmlDocument();
+			    nuspecXmlDocument.LoadXml(packageFile.Value.Contents);
+			    SetPackageDepencyVersionsToBeExplicitForXmlDocument(nuspecXmlDocument, dependencyNugetId);
+			    var updatedNuspecXml = nuspecXmlDocument.OuterXml;
+			    packageMetaData[packageFile.Key].Contents = updatedNuspecXml;
+		    }
+	    }
+
+	    private static void SetPackageDepencyVersionsToBeExplicitForXmlDocument(XmlDocument nuspecXmlDocument, string nugetIdFilter)
+	    {
+		    WalkDocumentNodes(nuspecXmlDocument.ChildNodes, node =>
```

Comment:
Wow, I haven't seen System.Xml in new code for quite some time, rather than a single XPathSelectElements. It'll do the job though.

Reply:
Had issues with namespacing, rather than add a namespace table and use XPaths I though this would be more reillient to nuspec namespace changes.